netket/optimizer/qgt/qgt_onthefly.py

```diff
@@ -139,7 +130,7 @@ def onthefly_mat_treevec(
     else:
         ravel_result = False
 
-    vec = tree_cast(vec, S.params)
+    vec = nkjax.tree_cast(vec, S.params)
 
```

Comment:
and here we should probably error/warn/implement it for anything other than the types of params
and not just silently cast. 
(same below)

Maybe in a separate PR though.

Reply:
Why should we error for different types of params?
I think it's still well defined, no?
If I convert the matrix to a dense matrix, then i can multiply it by a vector, whatever it's dtype.
Why can't i do it in the matmul then?